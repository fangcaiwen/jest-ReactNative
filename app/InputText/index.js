/**
 * Created by 208439 on 2019/4/3
 *
 * Author: wind
 *
 * Content:输入
 */

import React, { Component } from 'react';
import {
    StyleSheet,
    View,
    TextInput
} from 'react-native';

export default class InputText extends Component {
    constructor(props) {
        super(props);
        this.state = {
            value:props.value
        }
    }

    render() {
        const { value } = this.state;
        const { callBack } = this.props;
        return (
            <View style={styles.container}>
                <TextInput
                    style={styles.textInputStyle}
                    placeholder={'请输入您所需要的字符'}
                    onChangeText={(text) => callBack(text)}
                    value={value}
                    />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        width:300,
        height:50,
        borderWidth:1,
        borderColor:'#888',
        justifyContent:'center',
        borderRadius:4,
        marginBottom:10
    },
    textInputStyle:{
        flex:1,
        fontSize:16,
        paddingLeft:10
    }
});