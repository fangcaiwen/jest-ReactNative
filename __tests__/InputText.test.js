/**
 * Created by 208439 on 2019/4/3
 *
 * Author: wind
 *
 * Content:输入框测试
 */

import React from 'react';
import {configure, shallow,mount} from 'enzyme';
import InputText from '../app/InputText/index';
import renderer from "react-test-renderer";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Testing TextInput component', () => {

    // 测试UI
    it('renders TextInput as expected', () => {
        const tree = renderer.create(<InputText />).toJSON();
        // 快照测试
        expect(tree).toMatchSnapshot();
    });

    // 测试输入的默认值和调用改变方法
    it('renders TextInput as state', () => {
        const str = '哈哈';
        const wrapper = shallow(
            <InputText value={str} callBack={jest.fn((text) => text)} />
        );

        // 测试初始值是否成功赋予state
        expect(wrapper.state()).toEqual({value:str});

        // 测试初始值是否成功赋予到value
        expect(wrapper.props().children.props.value).toBe(str);

        // 测试填写input框，验证回调函数里面得到的值
        expect(wrapper.props().children.props.onChangeText('shshshsh')).toBe('shshshsh');
    });

});