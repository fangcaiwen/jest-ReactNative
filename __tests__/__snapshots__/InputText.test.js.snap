// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TextInput component renders TextInput as expected 1`] = `
<View
  style={
    Object {
      "borderColor": "#888",
      "borderRadius": 4,
      "borderWidth": 1,
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 10,
      "width": 300,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="请输入您所需要的字符"
    style={
      Object {
        "flex": 1,
        "fontSize": 16,
        "paddingLeft": 10,
      }
    }
  />
</View>
`;
